routing/payment_lifecycle.go

```diff
@@ -40,16 +41,19 @@ type paymentLifecycle struct {
 	quit chan struct{}
 
 	// resultCollected is used to signal that the result of an attempt has
-	// been collected. A nil error means the attempt is either successful
-	// or failed with temporary error. Otherwise, we should exit the
-	// lifecycle loop as a terminal error has occurred.
-	resultCollected chan error
+	// been collected.
+	resultCollected chan struct{}
 
 	// resultCollector is a function that is used to collect the result of
 	// an HTLC attempt, which is always mounted to `p.collectResultAsync`
 	// except in unit test, where we use a much simpler resultCollector to
 	// decouple the test flow for the payment lifecycle.
 	resultCollector func(attempt *channeldb.HTLCAttempt)
+
+	// switchResults is a map that holds the results for HTLC attempts
+	// returned from the htlcswitch.
+	switchResults lnutils.SyncMap[*channeldb.HTLCAttempt,
```

Comment:
Generally, it isn't safe to key by a _pointer_ to a struct. Unless this map is _only_ used to iterate over the contents and never a random look up. We've had prior very subtle bugs due to keying by a pointer.

Reply:
Good point! This is now removed.